src/test/scala/chiselTests/experimental/DataView.scala

```diff
@@ -554,6 +554,33 @@ class DataViewSpec extends ChiselFlatSpec {
     verilog should include("assign z = sel ? b : d;")
   }
 
+  it should "support muxing between views of Bundles" in {
+    import SimpleBundleDataView._
+    class MyModule extends Module {
+      val cond = IO(Input(Bool()))
+      val in1 = IO(Input(new BundleA(8)))
+      val in2 = IO(Input(new BundleA(8)))
+      val out = IO(Output(new BundleB(8)))
+
+      out := Mux(cond, in1.viewAs[BundleB], in2.viewAs[BundleB])
+    }
+    val verilog = ChiselStage.emitSystemVerilog(new MyModule, args = Array("--full-stacktrace"))
+    verilog should include("assign out_bar = cond ? in1_foo : in2_foo;")
+  }
+
+  it should "handle Probe of a view of a Bundle" in {
+    import SimpleBundleDataView._
+    class MyModule extends Module {
+      val in = IO(Input(new BundleA(8)))
+      val out_probe = IO(Output(Probe(new BundleB(8))))
+      val view = in.viewAs[BundleB]
+      define(out_probe, ProbeValue(view))
+    }
+    val verilog = ChiselStage.emitSystemVerilog(new MyModule)
+    verilog should include("wire [7:0] in_foo_probe = in_foo;")
+    verilog should include("`define ref_MyModule_out_probe_bar in_foo_probe")
```

Comment:
Just curious, why checking the SV and not the emitted CHIRRTL?

That said, I often find myself reaching for basically both -- check the emitted code is as-expected and also ensure it builds to SV at all or with some basic detail right.  I don't know, like I said just curious :+1:.

Reply:
To @seldridge's point about extra wires... emitted SV is a little neater rather than having to deal with intermediate wires, especially in this case when the wire gets a crappy name like `_WIRE` (maybe I can improve that...).

I will be a little more intentional about the wires and then check that they exist when they should (like now).

Finishing and merging https://github.com/chipsalliance/chisel/pull/3410 so that the tests could use FileCheck would help.

Reply:
Fixed!